Fix potential crash in AttributeName::tangoHostWithDomain()

bourtemb opened the pull request:
Fix crash when getaddrinfo is returning several struct addrinfo
(https://github.com/tango-controls-hdbpp/hdbpp-cm/issues/6#issuecomment-557124799)
The second struct addrinfo contains a field ai-canonname which is NULL.
This could cause a crash because some code is attempting to create a string from a null pointer.

getaddrinfo man page (on Debian Buster) says the following:

        If hints.ai_flags includes the AI_CANONNAME flag, then the ai_canon-
        name field of the first of the addrinfo structures in the returned
        list is set to point to the official name of the host.

So there is no need to look at the following addrinfo returned structures.

bourtemb commented:
I created this new Pull Request replacing https://github.com/tango-controls-hdbpp/libhdbpp-timescale/pull/7
This one will hopefully be merged in v0.11.0 branch as suggested by @chedburgh so the fix will be available in the next version.
This PR takes into account the recommendations from @gscalamera proposed in https://github.com/tango-controls-hdbpp/libhdbpp-timescale/pull/7

chedburgh commented:
Please add a note to the CHANGELOG.md under Unreleased.

bourtemb commented:
> Please add a note to the CHANGELOG.md under Unreleased.

Good point!
Done.

chedburgh reviewed (approved):
Looks good to me.

chedburgh commented:
One more change, can you space out the if statements please, so its clear they are a series of if's and not if/else etc.

bourtemb committed:
Fix potential crash in AttributeName::tangoHostWithDomain()
Fix crash when getaddrinfo is returning several struct addrinfo
(tango-controls-hdbpp/hdbpp-cm#6 (comment))
The second struct addrinfo contains a field ai-canonname which is NULL.
This could cause a crash because some code is attempting to create a string from a null pointer.

getaddrinfo man page (on Debian Buster) says the following:

        If hints.ai_flags includes the AI_CANONNAME flag, then the ai_canon-
        name field of the first of the addrinfo structures in the returned
        list is set to point to the official name of the host.

So there is no need to look at the following addrinfo returned structures.

bourtemb committed:
Prepare v0.11.1

bourtemb commented:
> One more change, can you space out the if statements please, so its clear they are a series of if's and not if/else etc.

Done. Thanks for the review.
